nav2_bt_navigator/behavior_trees/navigate_carrot.xml

```diff
@@ -0,0 +1,17 @@
+<!--
+  This Behavior Tree replans the global path periodically at 1 Hz.
+-->
+
+<root main_tree_to_execute="MainTree">
+  <BehaviorTree ID="MainTree">
+    <PipelineSequence name="NavigateWithReplanning">
+      <RateController hz="1.0">
+        <PipelineSequence name="GeneratePath">
+          <UpdateGoalPose goal="{goal}"/>
+          <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased"/>
```

Comment:
Do you not also want a replanning rate in case your haven't gotten an updated dynamic goal pose recently?